mne/gui/_ieeg_locate_gui.py

```diff
@@ -539,6 +542,42 @@ def _group_channels(self, groups):
                     base_names[base_name] = i
                     i += 1
 
+    def _update_lines(self, group, only_2D=False):
+        """Draw lines that connect the points in a group."""
+        if not only_2D and group in self._lines:
+            self._renderer.plotter.remove_actor(self._lines[group])
+        pos = np.array([
+            self._chs[ch] for i, ch in enumerate(self._ch_names)
+            if self._groups[ch] == group and i in self._seeg_idx and
+            not np.isnan(self._chs[ch]).any()])
+        if len(pos) < 2:  # not enough points for line
+            return
+        # first, the insertion will be the point farthest from the origin
+        # brains are a longer posterior-anterior, scale for this (80%)
+        insert_idx = np.argmax(np.linalg.norm(pos * np.array([1, 0.8, 1]),
+                                              axis=1))
+        # second, find the farthest point from the insertion
+        target_idx = np.argmax(cdist(pos, pos)[insert_idx])
```

Comment:
This should really use `pdist` rather than cdist so you can do (slightly less than) half the computations... but even then, you're doing `(n_ch * (n_ch - 1)) // 2` norms just to subselect one row (`n_ch` norms). I think this will only do the one row you care about:
```
        target_idx = np.argmax(cdist(pos[insert_idx][np.newaxis], pos))
```
Or you could avoid `cdist` and just do:
```
        target_idx = np.argmax(np.linalg.norm(pos[insert_idx] - pos, axis=1))
```
the `cdist` result will be of shape `(1, n_chs)` but that's fine because `argmax` operates on the raveled array

Reply:
Good catch!